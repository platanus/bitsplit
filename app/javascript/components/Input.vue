<template>
  <input
    class = "txt-input"
    :class= "[classmod, 'appearance-none border rounded w-full py-2 px-3  leading-tight focus:outline-none focus:shadow-outline']"
    :id="fieldId"
    :type="fieldType"
    :placeholder="fieldPlaceholder"
    :name="fieldName"
    :value="value"
    @input="$emit('update', $event.target.value)"
  >
</template>
<script>
export default {
  name: 'InputForm',
  props: {
    value: String,
    fieldId: String,
    fieldType: String,
    fieldPlaceholder: String,
    fieldName: String,
    classmod: {
      type: String,
      default: '',
    },
  },
  model: {
    prop: 'value',
    event: 'update',
  },
};
</script>
