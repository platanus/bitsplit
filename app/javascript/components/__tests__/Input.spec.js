import { shallowMount } from '@vue/test-utils'
import Input from '../Input'

describe('Input', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Input, {
      propsData: {
        fieldName: 'name',
        fieldFor: 'for',
        fieldPlaceholder: 'placeholder',
        fieldId: 'id'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
