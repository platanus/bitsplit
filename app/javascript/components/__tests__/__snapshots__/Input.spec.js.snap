// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `<input id="id" placeholder="placeholder" name="name" class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" fieldfor="for">`;
